tests/main/cgroup-tracking/task.yaml

```diff
@@ -0,0 +1,141 @@
+summary: Each snap app and hook is tracked via cgroups
+systems:
+    # Ubuntu 14.04's special version of systemd doesn't have StartTransientUnit API.
+    # This is covered in more depth in the sister cgroup-tracking-failure test.
+    - -ubuntu-14.04-*
+
+details: |
+    This test creates a snap process that suspends itself and ensures that it
+    placed into the appropriate hierarchy.
+environment:
+    USER/root: root
+    USER/test: test
+prepare: |
+    # This feature depends on the release-app-awareness feature
+    snap set core experimental.refresh-app-awareness=true
+
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    install_local test-snapd-sh
+
+    tests.session -u "$USER" prepare
+
+restore: |
+    snap unset core experimental.refresh-app-awareness
+    tests.session -u "$USER" restore
+    rm -f /tmp/*.pid /tmp/*.stamp
+    if [ "$USER" = root ]; then
+        systemctl --user stop dbus.service || true
+    fi
+
+execute: |
+    # TODO: add a variant that uses a service
+    if [ "$USER" = test ] && "$TESTSTOOLS"/version-compare --strict "$(systemctl --version | head -n 1 | cut -d ' ' -f 2)" -lt 238; then
+        echo "Systemd running as user session manager cannot start transient units"
+        echo "that cross ownership boundary. This test effectively runs through root"
+        echo "that is using ssh to connect to a test machine. Crossing the boundary"
+        echo "of cgroup ownership is allowed by systemd 238 or newer, with a new"
+        echo "dbus API offered by systemd running as system manager, which allows"
+        echo "overcoming this kernel limitation".
+        echo "SKIP: systemd is too old to perform this test as user"
+        exit 0
+    fi
+    if [ "$USER" = test ]; then
+        case "$SPREAD_SYSTEM" in
+            ubuntu-core-*)
+                echo "On core systems there is no session bus so the test variant does not make much sense"
+                echo "Please see the cgroup-tracking-failure test for more extensive analysis"
+                echo "SKIP: core systems don't have session services / session bus"
+                exit 0
+                ;;
+        esac
+    fi
+
+    # This test varies between Ubuntu 16.04, Ubuntu 18.04 and Fedora 31.
+    # This combination exercises each of the three cases below, namely:
+    # - pure cgroup v2 system, like Fedora 31
+    # - hybrid cgroup system, like Ubuntu 18.04
+    # - pure cgroup v1 system, like Ubuntu 16.04
+    echo "Find the path and id of the cgroup snapd uses for tracking."
+    if grep -q -F "0::" < /proc/self/cgroup; then
+        # Hybrid or pure v2 mode available.
+        if [ "$(stat -f --print=%T /sys/fs/cgroup)" = "cgroup2fs" ]; then
```

Comment:
```
. "$TESTSLIB"/systems.sh
if is_cgroupv2 ; then
   ...
```

maybe we could add a helper for is_cgroupv1_only too?

Reply:
Because there are three possible cases I wanted to make it evident how this is measured. I can add another helper but not sure it makes sense as relatively few tests actually need to measure this.